src/plugins/intel_cpu/tests/functional/shared_tests_instances/core_config.cpp

```diff
@@ -23,11 +23,6 @@ namespace ov {
 namespace test {
 
 void core_configuration(ov::test::SubgraphBaseTest* test) {
-    #if defined(OPENVINO_ARCH_X86) || defined(OPENVINO_ARCH_X86_64)
-        if (!test->configuration.count(InferenceEngine::PluginConfigParams::KEY_ENFORCE_BF16)) {
-            test->configuration.insert({InferenceEngine::PluginConfigParams::KEY_ENFORCE_BF16, InferenceEngine::PluginConfigParams::NO});
-        }
-    #endif
```

Comment:
Why remove it?

Reply:
for CPU plugin test, this is no need I think.

Reply:
@maxnick please, take a look

Reply:
`KEY_ENFORCE_BF16` property is not relevant for API 2.0 anymore. But, the check itself still makes sense. We have to force fp32 by default, as on some platfroms tests may be run in bf16 unexpectedly. Thus, if the inference precision is not specified in the config, we enforce fp32. Just like on lines 33-35.

Reply:
@maxnick @vurusovs Pls take a look, modified the if conditional.

Reply:
@maxnick do this check & force to f32, will get following test failed:
![image](https://github.com/openvinotoolkit/openvino/assets/52686861/56f5571e-ce65-4c11-983c-ab285f5755c0)
I'm debugging, any suggestions?

Reply:
@maxnick dump the graph & compare, seems here should not force to f32. Pls take a look:
![image](https://github.com/openvinotoolkit/openvino/assets/52686861/f08a9291-ec47-4fbd-94a6-6dc8cec944d2)